doc/nrf/app_dev/bootloaders: update about Mcuboot's LOAD_ADDRESS_CHECK (#29857)
nrfconnect/sdk-nrf: doc/nrf/app_dev/bootloaders_dfu/mcuboot_nsib/bootloader_main_config.rst

* ``CONFIG_UPDATEABLE_IMAGE_NUMBER`` - Specifies the number of images that MCUboot can handle.
* :kconfig:option:`SB_CONFIG_MCUBOOT_UPDATEABLE_IMAGES` - Configures the number of images supported by MCUboot at the sysbuild level.

Supported image matching modes
******************************

MCUboot supports various image matching modes that dictate how it recognizes the image target partition for an image candidate.
This feature is especially useful when MCUboot is used to update multiple images from the same incoming image slot, as is done for s0/s1 MCUboot instance updates.

> The choice of mode determines the capabilities of the bootloader's operations for the images supported by a given MCUboot instance:

Reviewer: ```suggestion
MCUboot supports various *image matching modes* that determine how it identifies the target partition for a given image candidate.

This feature is particularly useful when MCUboot updates multiple images from a single incoming image slot, as is the case for S0/S1 MCUboot instance updates.

The selected mode determines which bootloader operations are available for the images managed by a given MCUboot instance:```

.. list-table:: MCUboot image matching modes
    :header-rows: 1
    :widths: auto

    * - **Mode**
      - **Description**
      - **Configuration Option**
      - **Notes**
    * - Cortex-M reset handler-based (heuristic)

>       - Matches the image by checking whether the Cortex-M reset handler address in the image's vector table matches the partition address. This approach is a heuristic, is deprecated, and cannot be used with encrypted or compressed images.

Reviewer: ```suggestion
      - Matches the image by checking whether the Cortex-M reset handler address in the image's vector table matches the partition address. This approach is heuristic-based, is deprecated, and cannot be used with encrypted or compressed images.
```

>       - Matches the image by checking whether the Cortex-M reset handler address in the image's vector table matches the partition address. This approach is a heuristic, is deprecated, and cannot be used with encrypted or compressed images.

Reviewer 2: ```suggestion
      - Matches the image by comparing the Cortex-M reset handler address in the image's vector table against the partition address. 
      Because this approach is a heuristic, it is deprecated and cannot be used with encrypted or compressed images.
```

      - ``CONFIG_MCUBOOT_VERIFY_IMG_ADDRESS``
      - Deprecated; not compatible with encryption or compression.
    * - Explicit image address in header
      - Matches using an explicit image address encoded in the image header, set by the signing script. All executable slot addresses must be non-overlapping.
      - ``CONFIG_MCUBOOT_CHECK_HEADER_LOAD_ADDRESS``
      - Preferred for multi-image and updateable bootloader configurations such as s0/s1.
    * - Vendor and class UUID-based

>       - Matches images by the vendor and class UUIDs, which are stored in the image header. This approach allows reliable identification of images across devices and variants and is suitable for flexible multi-image systems. The configuration is provided in the device tree by a node with compatible ``nordic,mcuboot``, combined with the presence of ``uuid-vid-required`` and ``uuid-cid-required`` properties and boot chain descriptions.

Reviewer: ```suggestion
      - Matches images by the vendor and class UUIDs, which are stored in the image header. This approach allows reliable identification of images across devices and variants and is suitable for flexible multi-image systems. The configuration is provided in the devicetree by a node with compatible ``nordic,mcuboot``, combined with the presence of ``uuid-vid-required`` and ``uuid-cid-required`` properties and boot chain descriptions.
```

>       - Matches images by the vendor and class UUIDs, which are stored in the image header. This approach allows reliable identification of images across devices and variants and is suitable for flexible multi-image systems. The configuration is provided in the device tree by a node with compatible ``nordic,mcuboot``, combined with the presence of ``uuid-vid-required`` and ``uuid-cid-required`` properties and boot chain descriptions.

Reviewer 2: ```suggestion
      - Matches images by the vendor and class UUIDs, which are stored in the image header. 
       This approach allows reliable identification of images across devices and variants and is suitable for flexible multi-image systems. 
       The configuration is provided in the devicetree by a node with compatible ``nordic,mcuboot``.
       This node includes the ``uuid-vid-required`` and ``uuid-cid-required`` properties along with boot chain descriptions.
```

      - Device tree: node with ``nordic,mcuboot`` compatible

>       - Enables advanced matching via device tree properties for precise control across complex update scenarios.

Reviewer: ```suggestion
      - Enables advanced matching via devicetree properties for precise control across complex update scenarios.
```

    * - None

>       - No image matching is performed, nor is it checked whether the image candidate is expected to update the given image slot.

Reviewer 2: ```suggestion
      - The system does not perform image matching or verify whether the candidate image is expected to update the given image slot.
```
